apps/expo/app/post/[id].tsx

```diff
@@ -1,27 +1,33 @@
 import { SafeAreaView, Text, View } from "react-native";
-import { SplashScreen, Stack, usePathname } from "expo-router";
+import { SplashScreen, Stack, useSearchParams } from "expo-router";
 
 import { api } from "../../src/utils/api";
 
-const About: React.FC = () => {
-  const pathname = usePathname();
-  const id = pathname.split("/")[2] as string;
-  const { data } = api.post.byId.useQuery(id);
+const Post: React.FC = () => {
+  const { id } = useSearchParams();
+  const { data } = api.post.byId.useQuery(id || "", { enabled: !!id });
```

Comment:
see, this is just dumb that we need to validate that this exists...

Reply:
```suggestion
  if (!id) throw new Error("unreachable");
  const { data } = api.post.byId.useQuery(id);
```

Reply:
> see, this is just dumb that we need to validate that this exists...

Yeah, I don't see the props approach in docs. Maybe another thing for an issue? I have to make sure it doesn't exist though